packages/vscode-extension/src/devices/DeviceNameCache.ts

```diff
@@ -0,0 +1,141 @@
+import path from "path";
+import { workspace, Uri } from "vscode";
+import { extensionContext } from "../utilities/extensionContext";
+import { Logger } from "../Logger";
+
+type DeviceModel = {
+  brand: string;
+  name: string;
+};
+type DeviceModels = Record<string, DeviceModel>;
+
+type CacheFile = {
+  savedAt: number;
+  data: DeviceModels;
+};
+
+const DEVICE_MODELS_URL = "https://cdn.jsdelivr.net/gh/bsthen/device-models/devices.json";
+const DEVICE_MODELS_FILENAME = "RNIDE_device_models.json";
+const CACHE_DURATION_MS = 5 * 24 * 60 * 60 * 1000;
+
+async function readCacheFile(): Promise<CacheFile | null> {
+  try {
+    const filePath = path.join(extensionContext.globalStorageUri.fsPath, DEVICE_MODELS_FILENAME);
+    const fileUri = Uri.file(filePath);
+    const file = await workspace.fs.readFile(fileUri);
+    const json = JSON.parse(file.toString()) as CacheFile;
+    return json;
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException)?.code === "ENOENT") {
+      Logger.info("Device models cache file does not exist, a new one will be created");
+      return null;
+    }
+    Logger.error(`Error reading device models cache file: ${(error as Error).message}`);
+    return null;
+  }
+}
+
+async function writeCacheFile(data: DeviceModels): Promise<void> {
+  try {
+    const filePath = path.join(extensionContext.globalStorageUri.fsPath, DEVICE_MODELS_FILENAME);
```

Comment:
Same as in the other function, you can use `Uri.joinPath` instead of extracting the path and creating a new URI.